src/math/vector.ts

```diff
@@ -954,21 +954,15 @@ export function color4FromHex4(hex: number, out?: Vec4) {
 
 export function slerpQuat(from: Vec4, to: Vec4, t: number, out?: Vec4): Vec4 {
   out = out || [0, 0, 0, 0];
-  const to1: Vec4 = [0, 0, 0, 0];
   let omega, cosom, sinom, scale0, scale1;
   cosom = from[1] * to[1] + from[2] * to[2] + from[3] * to[3] + from[0] * to[0];
 
   if (cosom < 0.0) {
     cosom = -cosom;
-    to1[0] = -to[1];
-    to1[1] = -to[2];
-    to1[2] = -to[3];
-    to1[3] = -to[0];
-  } else {
-    to1[0] = to[1];
-    to1[1] = to[2];
-    to1[2] = to[3];
-    to1[3] = to[0];
+    to[0] = -to[0];
+    to[1] = -to[1];
+    to[2] = -to[2];
+    to[3] = -to[3];
```

Comment:
It's not valid to modify an input like this. This causes unpredictability in the code.

```
const start = [1, 2, 3, 4];
const end = [4, 3, 2, 1];
const result = slerpQuat(start, end, 0.6);
console.log(end); // This outputs [-4, -3, -2, -1] (if cosom is less than zero) There's nothing about the method that indicates this is expected behavior, so it could be a major problem if I intend to keep using "end"
```